src/external/wannier_shared.jl

```diff
@@ -0,0 +1,407 @@
+# Shared functionality to operate with Wannier programs (Wannier.jl and Wannier90).
+# The communication happens with the Wannier90 file formats which Wannier.jl also understands.
+
+using Dates
+
+@doc raw"""
+Base type for Wannier projections.
+
+Wannierization searchs for a unitary matrix ``U_{m_n}``.
+As a starting point for the search, we can provide an initial guess function ``g``
+for the shape of the Wannier functions, based on what we expect from knowledge of the problem or physical intuition.
+This starting matrix is called ``[A_k]_{m,n}``, and is computed as follows:
+``[A_k]_{m,n} = \langle ψ_m^k | g^{\text{per}}_n \rangle``.
+The matrix will be orthonormalized by the chosen Wannier program, we don't need to do so ourselves.
+"""
+abstract type WannierProjection end
```

Comment:
I don't think this abstract type is needed. Let me know if you have reason to believe otherwise.